Week01/Problem04/utantipongpipat_04.py

```diff
@@ -0,0 +1,20 @@
+import math
+
+def parTest(number):
+    strIn = str(number);
+    for i in range (math.floor(len(strIn)/2)):
```

Comment:
Instead of iterating over each character in the strings and checking that each pair of characters match individually, you could reverse the order of the string (i.e., `strIn[::-1]`) and compare the strings with `strIn == strIn[::-1]`.  This should reduce the algorithmic complexity (which, admittedly, we haven't talked about yet) by removing the need for your loop here, which in turn will make your code run faster.  

Try this.  Use the `%timeit` magic function in a Jupyter notebook cell to time the following code.
Cell 01: Defining functions to time
```
# This one is your parTest function
def loop_parTest(number):
    """Compares string repr's of integers character-by-character
    to determine if the integer is a palindrome"""
    strIn = str(number);
        for i in range (math.floor(len(strIn)/2)):
            if (strIn[i] != strIn[len(strIn)-i-1]):
                return False
            else: 
                return True
# This compares the strings directly
def noloop_parTest(number):
    """Compares string repr's of integers directly to determine
    if the integer is a palindrome"""
    if str(number) == str(number)[::-1]:
        return True
    else: 
        return False
```
Cell 02: Timing your parTest function
```
%timeit
number = 134098350961023985
loop_parTest(number)
```
Cell 03: Time no-loop function
```
%timeit
number = 134098350961023985
noloop_parTest(number)
```